### PR TITLE
Added legacy fields to task-api comp.task rpc command

### PR DESCRIPTION
**Pull request by maaktweluit**
This way electron does not crash after adding a task-api task.

### CODE REVIEW

**Comment by kmazurek** [comment]
I think you should change the base branch here to `b0.22`.

**Comment by maaktweluit** [comment]
@zakaprov you are correct, thanks!

Updated the base

File: golem/client.py [hunk 1]
```diff
@@ -897,14 +899,59 @@ def get_task(self, task_id: str) -> Optional[dict]:
...
                 'id': task.task_id,
+                'time_remaining': time_remaining,
+                'subtasks_count': task.max_subtasks,
                 'status': task.status.value,
+                'progress': progress,
                 'time_started': time_started,
+                'last_updated': last_updated,
+                'name': task.name,
+                'bid': float(task.max_price_per_hour) / denoms.ether,
+                'compute_on': task.env_id,
```
Comment (mfranciszkiewicz) [inline golem/client.py]: Afair this should be either cpu or gpu

File: golem/client.py [hunk 2]
```diff
@@ -897,14 +899,59 @@ def get_task(self, task_id: str) -> Optional[dict]:
             if not task:
                 return None
             subtask_ids = rtm.get_requested_task_subtask_ids(task_id)
+            # time_started
             if task.start_time is None:
                 time_started = get_timestamp_utc()
+                if task.status == taskstate.TaskStatus.errorCreating:
+                    time_started -= 1
```
Comment (mfranciszkiewicz) [inline golem/client.py:907]: I believe you meant `time_started = -1`. Maybe use `None` if it’s handled properly?

File: golem/client.py (hunk 1 above)
Comment (maaktweluit) [reply golem/client.py]: do we want to match the values for old tasks? i think `docker_cpu` and `docker_gpu` describe better where the task is computed on.

In my tests this was not breaking the GUI

@mdtanrikulu what do you think?

File: golem/client.py (hunk 2 above)
Comment (maaktweluit) [reply golem/client.py:907]: the idea was to show tasks in the order:
- Tasks that started
- errorCreating
- creating

When both creating and errorCreating get assigned to `now()` there is no order

`None` would be best but i'm not sure if its handled properly, that would be a fix for develop

File: golem/client.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/client.py]: If the goal is to add legacy fields, then I believe we do

Comment (maaktweluit) [reply golem/client.py]: looking at electron this value is only printed: https://github.com/golemfactory/golem-electron/search?q=compute_on&unscoped_q=compute_on

my proposal is to extend the possible values of this field with the new envs:
- `cpu` -> non task-api cpu
- `docker_cpu` -> task-api cpu
- etc...

this means we do not have to store a mapping from `env_id` to `compute_on`

why do you think its better to only stick with the old values? happy to change it :)

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**